vendor/github.com/openshift-kni/performance-addon-operators/functests/4_latency/latency.go

```diff
@@ -0,0 +1,257 @@
+	AfterEach(func() {
+		var err error
+		err = testclient.Client.Delete(context.TODO(), oslatPod)
+		if err != nil {
+			klog.Error(err)
+		}
+		err = pods.WaitForDeletion(oslatPod, 60*time.Second)
+		if err != nil {
+			klog.Error(err)
+		}
+	})
+
+	Context("with the oslat image", func() {
+		It("should succeed", func() {
+			oslatPod = getOslatPod(profile, workerRTNode)
+			err := testclient.Client.Create(context.TODO(), oslatPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			timeout, err := strconv.Atoi(latencyTestRuntime)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting two minutes to download the oslat image")
+			err = pods.WaitForPhase(oslatPod, corev1.PodRunning, 2*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting another two minutes to give enough time for the cluster to move the pod to Succeeded phase")
+			podTimeout := time.Duration(timeout + 120)
+			err = pods.WaitForPhase(oslatPod, corev1.PodSucceeded, podTimeout*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			cmd := []string{"cat", "/rootfs/var/log/oslat.log"}
+			out, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			// verify the maximum latency only when it requested, because this value can be very different
+			// on different systems
+			if maximumLatency == -1 {
+				return
+			}
+
+			maximumRegex, err := regexp.Compile(`Maximum:\t*\s*(.*)\s*\(us\)`)
+			Expect(err).ToNot(HaveOccurred())
+
+			latencies := maximumRegex.FindSubmatch([]byte(out))
+			Expect(maximumLatency).ToNot(BeNil())
+
+			// under the output of the oslat very often we have one anomaly high value, for example
+			// Maximum:    16543 15 15 14 13 12 12 13 12 12 12 12 12 12 12 12 12 (us)
+			// it still unclear if it oslat bug or the kernel one, but we definitely do not want to
+			// fail our test on it
+			var anomaly bool
+			for _, lat := range strings.Split(string(latencies[1]), " ") {
+				if lat == "" {
+					continue
+				}
+
+				curr, err := strconv.Atoi(lat)
+				Expect(err).ToNot(HaveOccurred())
+
+				// skip the anomaly value
+				if curr > maximumLatency && !anomaly {
+					anomaly = true
+					continue
+				}
+
+				Expect(curr < maximumLatency).To(BeTrue(), fmt.Sprintf("The current latency %d is bigger than the expected one %d", curr, maximumLatency))
+			}
+		})
+	})
+})
+
+func getOslatPod(profile *v1.PerformanceProfile, node *corev1.Node) *corev1.Pod {
+	cpus := cpuset.MustParse(string(*profile.Spec.CPU.Isolated))
+	runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+	volumeTypeDirectory := corev1.HostPathDirectory
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "oslat-",
+			Annotations: map[string]string{
+				"cpu-load-balancing.crio.io": "true",
+			},
+			Namespace: testutils.NamespaceTesting,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy:    corev1.RestartPolicyNever,
+			RuntimeClassName: &runtimeClass,
+			Containers: []corev1.Container{
```

Comment:
Should this call the oslat-run.sh command?

Reply:
Currently, it as `oslat-run.sh` as entry point, so no need to specify the command

Reply:
So this is gonna change when you switch to cnf tests?

Reply:
yep, like under the test it will run cnf-tests image with the `oslat-runnet` command